src/api/VoteMonitor.Api.Form/CommandHandlers/DeleteFormCommandHandler.cs

```diff
@@ -0,0 +1,55 @@
+﻿using MediatR;
+using Microsoft.EntityFrameworkCore;
+using System;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using VoteMonitor.Api.Form.Commands;
+using VoteMonitor.Entities;
+
+namespace VoteMonitor.Api.Form.CommandHandlers
+{
+    public class DeleteFormCommandHandler : IRequestHandler<DeleteFormCommand, bool>
+    {
+        private readonly VoteMonitorContext _context;
+
+        public DeleteFormCommandHandler(VoteMonitorContext context)
+        {
+            this._context = context;
+        }
+
+        public async Task<bool> Handle(DeleteFormCommand request, CancellationToken cancellationToken)
+        {
+            var form = await _context.Forms.FirstOrDefaultAsync(f => f.Id == request.FormId);
+            if (form == null)
+            {
+                return false;
+            }
+
+            var sections = _context.FormSections.Where(s => s.IdForm == form.Id);
+            var sectionsIds = sections.Select(s => s.Id);
+            var questions = _context.Questions.Where(q => sectionsIds.Contains(q.IdSection));
+            var questionsIds = questions.Select(q => q.Id);
+            var optionsToQuestions = _context.OptionsToQuestions.Where(o => questionsIds.Contains(o.IdQuestion));
+            var optionsIds = optionsToQuestions.Select(o => o.IdOption);
+
+            // check if there are already saved answers
+            var answers = _context.Answers.Where(a => optionsIds.Contains(a.IdOptionToQuestion));
+            if (answers != null && answers.Any())
```

Comment:
`answers` is always not null here, suggest using AnyAsync as Any queries in a blocking way

Reply:
Right, I moved it from FormQuestionQueryHandler.cs without reviewing it. I removed it and I refactored this handler as well.